### PR TITLE
support python 3.9 in client and drop support for python 2.7 everywhere

### PR DESCRIPTION
**Pull request by bcavagnolo**
This work advances contributions from @lod and @janw.

### CODE REVIEW

File: api/datalake_api/querier.py [hunk 1]
```diff
@@ -222,6 +223,7 @@ def _add_cursor_conditions(self, kwargs, cursor):
             kwargs["ExpressionAttributeNames"]["#n2"] = "metadata"
             kwargs["ExpressionAttributeNames"]["#n3"] = "id"
             kwargs["ExpressionAttributeValues"][":v2"] = cursor.last_id
+            pass
```
Comment (oldpatricka) [inline api/datalake_api/querier.py:226]: Hmm, why is this necessary?

File: api/tests/conftest.py [hunk 2]
```diff
@@ -14,25 +14,58 @@
...
 from datalake.common import DatalakeRecord
+from datalake.tests import generate_random_metadata
 
 
 YEAR_2010 = 1262304000000
 
 
-@pytest.fixture
-def client():
+# If we run with proper AWS credentials they will be used
```
Comment (oldpatricka) [inline api/tests/conftest.py:32]: Nice.

File: api/tests/test_archive_querier.py [hunk 3]
```diff
@@ -249,10 +249,12 @@ def get_multiple_pages(query_function, query_args):
 def get_all_pages(query_function, query_args):
     pages = []
     cursor = None
+    counter = 1
     while True:
         page = get_page(query_function, query_args, cursor)
         pages.append(page)
         cursor = page.cursor
+        counter += 1
```
Comment (oldpatricka) [inline api/tests/test_archive_querier.py:257]: hmm am i missing where this counter is used? I don't see why its added.

File: api/datalake_api/querier.py (hunk 1 above)
Comment (bcavagnolo) [reply api/datalake_api/querier.py:226]: Ha. No. It's an artifact from some debugging. Good catch.

File: api/tests/conftest.py (hunk 2 above)
Comment (bcavagnolo) [reply api/tests/conftest.py:32]: Props to @lod for this technique.

File: api/tests/test_archive_querier.py (hunk 3 above)
Comment (bcavagnolo) [reply api/tests/test_archive_querier.py:257]: Derp. Also a debug artifact. Thanks.

**Review by oldpatricka — APPROVED** [review]
